README.md

```diff
@@ -1,15 +1,261 @@
+
+You can view your generated configuration using:
+
+```bash
+ackdev config view
+```
+
+The configuration file will look like:
+
+``` yaml
+rootDirectory: /home/amine/go/github.com/aws-controllers-k8s/dev-tools
+git:
+  sshKeyPath: ""
+github:
+  token: ""
+  username: ""
+  forkPrefix: ""
+repositories:
+  core:
+  - runtime
+  - dev-tools
+  - community
+  - code-generator
+  services:
+  - s3
+run:
+  flags: {}
+```
+
+To use all `ackdev` features you will need to fill the `git.sshKeyPath` and `github` sections.
+You can do that using the `ackdev config` command,
+
+For example to set the Github username you can run:
+
+```bash
+ackdev config set github.username a-hilaly
+```
+
+The `git.sshKeyPath` should point to the private key you use to push code to Github.
+The `github.token` should contain a token that provides `fork/renaming` rights (`repo/*` policies). You can create one by following these [instructions][github-token]
+
+[github-token]: https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
+
+### Dependencies
+
+`ackdev` can help you manage dependencies and tools you will need in your ACK development journey.
+To start you can run:
+
+```bash
+ackdev list dependencies # deps|dep|dependency [--short-path]
+```
+
+The output will look like:
+```bash
+NAME           STATUS    VERSION         PATH                     
+kind           OK        0.9.0           /usr/local/bin/kind      
+helm           OK        v3.2.4+g0ad800e /usr/local/bin/helm      
+mockery        NOT FOUND -                                        
+kubectl        OK        v1.20.0         /usr/local/bin/kubectl   
+kustomize      NOT FOUND -
+controller-gen OK        v0.4.0          /usr/bin/controller-gen
+```
+
+To install the missing dependencies you can run
+
+```bash
+ackdev ensure deps # dependencies||dep|dependency
+```
+
+```bash
+Installing mockery into bin/mockery ... ok.
+Installing kustomize from https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh ... ok.
+```
+
+### Repositories
+
+```bash
+ack list repositories # repo|repos|repository --show-url
+```
+
+The output should look like this
+```bash
+NAME                TYPE       BRANCH 
+runtime             core         
+community           core       docs
+code-generator      core         
+dev-tools           core        
+s3-controller       controller main
+sns-controller      controller bug-fix
+dynamodb-controller controller
+```
+
+To fork/clone all the listed repositories run:
+
+```bash
+ackdev ensure repos
+```
+
+## Examples
+
+### Running controllers
+
+Set run flags config
+```
+ackdev config set run.flags.aws-account-id $AWS_ACCOUNT_ID
+ackdev config set run.flags.aws-region $AWS_REGION
+ackdev config set run.flags.enable-development-logging true
+ackdev config set run.flags.log-level debug
```

Comment:
Continuing my conversation from above, you could have a well-documented configuration file that has YAML comments above each configuration toggle and instead of keeping this README up to date as you change various configuration options and add new ones, if you just had a single `ackdev config edit` command, you'd just need to modify the example YAML configuration file with the per-element YAML comments.